sips/committee_consensus.md

```diff
@@ -0,0 +1,486 @@
+}
+
+// FilterCommitteeDuty filters the committee duty. It returns the new duty, the validators to stop and the highest attesting slot map
+func FilterCommitteeDuty(duty *types.CommitteeDuty, slotMap map[types.ValidatorPK]spec.Slot) (
+	*types.CommitteeDuty,
+	map[spec.Slot]types.ValidatorPK,
+	map[types.ValidatorPK]spec.Slot) {
+	validatorsToStop := make(map[spec.Slot]types.ValidatorPK)
+
+	for i, beaconDuty := range duty.BeaconDuties {
+		validatorPK := types.ValidatorPK(beaconDuty.PubKey)
+		slot, exists := slotMap[validatorPK]
+		if exists {
+			if slot < beaconDuty.Slot {
+				validatorsToStop[beaconDuty.Slot] = validatorPK
+				slot = beaconDuty.Slot
+			} else { // else don't run duty with old slot
+				duty.BeaconDuties[i] = nil
+			}
+		}
+	}
+	return duty, validatorsToStop, slotMap
+}
+```
+2. `Committee` receives consensus messages and hands them over `CommitteeRunner` that hands them over to the `QBFTController` that has unchanged logic. The only difference is `BeaconVote` is used as the ConsensusData object.
+3.  Once `ProcessConsensus` decides, the `Committee` will create a post consensus of PartialSignatureMessage that aggregates the beacon partial signature for all relevant validators.
+4. `Committee` will process post-consensus partial signature messages and submit a beacon message for each validator.
+
+
+### Stopping Runs
+
+Previously we have letted new validator duties stop the run for the previous duty. Now we don't stop the runner but just mark certain validators as stopped and omit the partial sig emission for them.
+
+### Omitting Partial Signatures
+
+If in post-consensus stage for attestation duty, the duty's slot is lower than a validator's `highestAttestingSlot`, the `CommitteeRunner` will omit the post-consesnus message for this validator. 
+
+### Cutoff Round
+We will also create a better Cutoff Round.
+
+#### Sync Committee
+`CutOffRound = 4  \\ one slot`
+
+#### Attestation
+`CutOffRound = 12 \\ one epoch`
+
+### CommitteeID
+
+An identifier for committee must be added to `MessageID`.
+
+```go
+type CommitteeID [32]byte
+
+// Return a 32 bytes ID for the committee of operators
+func getCommitteeID(committee []OperatorID) CommitteeID {
+	// sort
+	sort.Slice(committee, func(i, j int) bool {
+		return committee[i] < committee[j]
+	})
+	// Convert to bytes
+	bytes := make([]byte, len(committee)*4)
+	for i, v := range committee {
+		binary.LittleEndian.PutUint32(bytes[i*4:], uint32(v))
+	}
+	// Hash
+	return CommitteeID(sha256.Sum256(bytes))
+}
+```
+
+In order to route consensus messages to the correct consensus runner, the `CommitteeID` field will be included in the `MessageID` replacing `ValidatorPublicKey`.
+
+### Role
+
+This is used to route the message to the correct runner. Since `CommitteeRunner` has several beacone roles we will create a new `RunnerRole` in the MessageID.
+
+```go
+// TODO since this is on wire no real need to take 32 bits
+type RunnerRole int32
```

Comment:
yes, this can be changed to int8, but if it's not a very trivial change, i'd not include it in this SIP, because i doubt this optimization would be even visible in any chart at all (be it traffic, memory usage, etc..)

Reply:
removing the TODO because there is a comment..
If we don't do this it should be added to #39

Reply:
for you it is trivial @moshe-blox ?

Reply:
at this point it's not worth the change imo